contracts/utils/AssociatedArrayLib.sol

```diff
@@ -10,45 +10,68 @@ library AssociatedArrayLib {
         uint256 _spacer;
     }
 
-    function _length(Array storage s, address account) private view returns (uint256 __length) {
+    function _slot(Array storage s, address account) private pure returns (bytes32 __slot) {
         assembly {
             mstore(0x00, account)
             mstore(0x20, s.slot)
-            __length := sload(keccak256(0x00, 0x40))
+            __slot := keccak256(0x00, 0x40)
+        }
+    }
+
+    function _length(Array storage s, address account) private view returns (uint256 __length) {
+        bytes32 slot = _slot(s, account);
+        assembly {
+            __length := sload(slot)
         }
     }
 
     function _get(Array storage s, address account, uint256 index) private view returns (bytes32 value) {
-        if (index >= _length(s, account)) revert AssociatedArray_OutOfBounds(index);
+        return _get(_slot(s, account), index);
+    }
+
+    function _get(bytes32 slot, uint256 index) private view returns (bytes32 value) {
         assembly {
-            mstore(0x00, account)
-            mstore(0x20, s.slot)
-            value := sload(add(keccak256(0x00, 0x40), mul(0x20, add(index, 1))))
+            //if (index >= _length(s, account)) revert AssociatedArray_OutOfBounds(index);
+            if iszero(lt(index, sload(slot))) {
+                mstore(0, 0x8277484f) // `AssociatedArray_OutOfBounds(uint256)`
+                mstore(4, index)
+                revert(0x1c, 0x24)
+            }    
+            value := sload(add(slot, mul(0x20, add(index, 1))))
         }
     }
 
     function _getAll(Array storage s, address account) private view returns (bytes32[] memory values) {
-        uint256 __length = _length(s, account);
+        bytes32 slot = _slot(s, account);
+        uint256 __length;
+        assembly {
+            __length := sload(slot)
+        }
         values = new bytes32[](__length);
         for (uint256 i; i < __length; i++) {
-            values[i] = _get(s, account, i);
+            values[i] = _get(slot, i);
         }
     }
 
     function _set(Array storage s, address account, uint256 index, bytes32 value) private {
-        if (index >= _length(s, account)) revert AssociatedArray_OutOfBounds(index);
+        _set(_slot(s, account), index, value);
+    }
+
+    function _set(bytes32 slot, uint256 index, bytes32 value) private {
         assembly {
-            mstore(0x00, account)
-            mstore(0x20, s.slot)
-            sstore(add(keccak256(0x00, 0x40), mul(0x20, add(index, 1))), value)
+            //if (index >= _length(s, account)) revert AssociatedArray_OutOfBounds(index);
+            if iszero(lt(index, sload(slot))) {
+                mstore(0, 0x8277484f) // `AssociatedArray_OutOfBounds(uint256)`
+                mstore(4, index)
+                revert(0x1c, 0x24)
+            }    
```

Comment:
Note that the first `mstore()` is writing to `[0x00, 0x20)`, and specifically since it's a 4-byte value, the `0x8277484f` will be placed in `[0x1c, 0x20)`. The second `mstore()` will write to `[0x04, 0x24)` which will clobber 4-byte value from the first `mstore()` and might not match the memory where the `revert()` happens. Recommendation:

Changing the second `mstore()` to write to `[0x20, 0x40)` would work:

```diff
if iszero(lt(index, sload(slot))) {
    mstore(0, 0x8277484f) // `AssociatedArray_OutOfBounds(uint256)`
-   mstore(4, index)
+   mstore(0x20, index)
    revert(0x1c, 0x24)
}
```